src/main.rs

```diff
@@ -1047,6 +1061,57 @@ mod tests {
         });
     }
 
+    #[test]
+    fn test_vunet() {
+        test_block!(tb, "", {
+            let mut clear = ClearDiskConfig::new();
+            let guest = Guest::new(&mut clear);
+
+            let mut workload_path = dirs::home_dir().unwrap();
+            workload_path.push("workloads");
+
+            let mut vubridge_path = workload_path.clone();
+            vubridge_path.push("vubridge");
+            let vubridge_path = String::from(vubridge_path.to_str().unwrap());
+
+            // Start the daemon
+            let mut daemon_child = Command::new(vubridge_path.as_str()).spawn().unwrap();
+
+            let mut cloud_child =
+                Command::new("/root/cloud-hypervisor/target/debug/cloud-hypervisor")
+                    .args(&["--cpus", "4"])
+                    .args(&["--memory", "size=512M,file=/dev/shm"])
+                    .args(&["--kernel", guest.fw_path.as_str()])
+                    .args(&[
+                        "--disk",
+                        guest
+                            .disk_config
+                            .disk(DiskType::OperatingSystem)
+                            .unwrap()
+                            .as_str(),
+                        guest
+                            .disk_config
+                            .disk(DiskType::CloudInit)
+                            .unwrap()
+                            .as_str(),
+                    ])
+                    .args(&["--vunet", "mac=52:54:00:02:d9:01,sock=/tmp/vubr.sock"])
+                    .args(&["--net", guest.default_net_string().as_str()])
+                    .spawn()
+                    .unwrap();
+
+            thread::sleep(std::time::Duration::new(20, 0));
+            let _ = daemon_child.kill();
+            let _ = daemon_child.wait();
+            thread::sleep(std::time::Duration::new(10, 0));
```

Comment:
I think you need to test the network itself.

Reply:
@bjzhjing Here you're only launching the VMM, then the vhost-user bridge, but you're not testing connectivity overall.
You should be able to at least ping the guest from the host once those 2 processes are running.

Reply:
Ok, I'd like to issue "ifconfig" with ssh_command to get TX package statistic data.

Reply:
Try the same command "ip -o link | wc -l" like test_multiple_network_interfaces,  to ensure there are 3 interfaces, default net + vhost_user_net interface+ local.